File: 1894-merge-strings-alternately/solution.go
func mergeAlternately(word1 string, word2 string) string {
    res := ""
    for i := 0; i < len(word1) || i < len(word2); i++ {
        if i < len(word1) {
            res += string(word1[i])
        }
        if i < len(word2) {
            res += string(word2[i])
        }
    }

    return res
}

File: 1894-merge-strings-alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = ""
        min_len = min(len(word1), len(word2))
        for i in range(min_len):
            res += word1[i]
            res += word2[i]
        
        if len(word1) > len(word2):
            res += word1[min_len:]
        elif len(word2) > len(word1):
            res += word2[min_len:]

        return res
